library/sinks/Prisma.test.ts

```diff
@@ -0,0 +1,342 @@
+    {
+      cwd: path.join(__dirname, "fixtures/prisma/sqlite"),
+    }
+  );
+
+  if (stderr) {
+    t.fail(stderr);
+  }
+
+  const { PrismaClient } = require("@prisma/client");
+
+  const client = new PrismaClient();
+
+  await client.user.create({
+    data: {
+      name: "Alice",
+      email: "alice@example.com",
+    },
+  });
+
+  t.same(await client.$queryRawUnsafe("SELECT * FROM USER"), [
+    {
+      id: 1,
+      name: "Alice",
+      email: "alice@example.com",
+    },
+  ]);
+
+  await runWithContext(context, async () => {
+    t.same(await client.$queryRawUnsafe("SELECT * FROM USER"), [
+      {
+        id: 1,
+        name: "Alice",
+        email: "alice@example.com",
+      },
+    ]);
+
+    try {
+      await client.$queryRawUnsafe("SELECT * FROM USER -- should be blocked");
+      t.fail("Query should be blocked");
+    } catch (error) {
+      t.ok(error instanceof Error);
+      if (error instanceof Error) {
+        t.same(
+          error.message,
+          "Zen has blocked an SQL injection: prisma.$queryRawUnsafe(...) originating from body.myTitle"
+        );
+      }
+    }
+  });
+
+  await client.$executeRawUnsafe("DELETE FROM USER WHERE id = 1");
+
+  await client.user.create({
+    data: {
+      name: "Alice2",
+      email: "alice2@example.com",
+    },
+  });
+
+  await runWithContext(context, async () => {
+    await client.$executeRawUnsafe("DELETE FROM USER WHERE id = 2");
+
+    try {
+      await client.$executeRawUnsafe("DELETE FROM USER WHERE id = 2");
+      await client.$executeRawUnsafe(
+        "DELETE FROM USER WHERE id = 1 -- should be blocked"
+      );
+      t.fail("Execution should be blocked");
+    } catch (error) {
+      t.ok(error instanceof Error);
+      if (error instanceof Error) {
+        t.same(
+          error.message,
+          "Zen has blocked an SQL injection: prisma.$executeRawUnsafe(...) originating from body.myTitle"
+        );
+      }
+    }
+
+    try {
+      await client.$executeRawUnsafe();
+      t.fail("Should not be reached");
+    } catch (error) {
+      t.ok(error instanceof Error);
+      if (error instanceof Error) {
+        t.match(
+          error.message,
+          /Invalid `prisma\.\$executeRawUnsafe\(\)` invocation/
+        );
+      }
+    }
+  });
+
+  await client.$disconnect();
+});
+
+t.test("it works with postgres", async (t) => {
+  const agent = createTestAgent();
+  agent.start([new Prisma()]);
+
+  process.env.DATABASE_URL = "postgres://root:password@127.0.0.1:27016/main_db";
+
+  // Generate prismajs client
+  const { stdout, stderr } = await execAsync(
+    "npx prisma migrate reset --force", // Generate prisma client, reset db and apply migrations
+    {
+      cwd: path.join(__dirname, "fixtures/prisma/postgres"),
+    }
+  );
+
+  if (stderr) {
+    t.fail(stderr);
+  }
+
+  // Clear require cache
+  for (const key in require.cache) {
+    delete require.cache[key];
+  }
```

Comment:
why are we doing this? to be able to load prisma again? maybe cleaner to put in a different test file?

Reply:
Yes, to load it again. Putting in different test files does not work because they are executed in parallel.